### PR TITLE
UIPFU-87 - Fix User Search Modal pagination issue.

### PR DESCRIPTION
**Pull request by Terala-Priyanka**
[UIPFU-87](https://folio-org.atlassian.net/browse/UIPFU-87) - Fix User Search Modal pagination issue.

The fix includes :-
-> displaying load more button when filtered data > 100.
-> showing correct filtered data count in sub-header (below User Search Results heading).
 
 ## Screencast
https://github.com/folio-org/ui-plugin-find-user/assets/105021655/e8ce5005-c381-4d8a-8fb9-370d847cd73a

## Unit tests
Test cases were not covered initially.
We will gradually migrate unit tests in this module to Jest/RTL .

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/UserSearchContainer.js [hunk 1]
```diff
@@ -159,7 +160,13 @@ class UserSearchContainer extends React.Component {
 
   onNeedMoreData = (askAmount, index) => {
     const { resultOffset } = this.props.mutator;
-
+    if(index/askAmount < 1) {
```
Comment (BogdanDenis) [inline src/UserSearchContainer.js:163]: This works, but I think simple `index < askAmount` is more understandable

Comment (artem-blazhko) [inline src/UserSearchContainer.js:163]: Could you fix ESLint errors here?

File: src/UserSearchContainer.js [hunk 2]
```diff
@@ -159,7 +160,13 @@ class UserSearchContainer extends React.Component {
...
+      /*
+        This condition sets offset to 100 when there are less than 100 records in the current
```
Comment (artem-blazhko) [inline src/UserSearchContainer.js:165]: Could you fix ESLint errors here?

**Review by BogdanDenis — CHANGES_REQUESTED** [review]

File: src/UserSearchContainer.js [hunk 3]
```diff
@@ -159,7 +160,13 @@ class UserSearchContainer extends React.Component {
 
   onNeedMoreData = (askAmount, index) => {
     const { resultOffset } = this.props.mutator;
-
+    if(index/askAmount < 1) {
+      /*
+        This condition sets offset to 100 when there are less than 100 records in the current 
+        paginated result in order to skip the first 100 records and make an API call to fetch next 100.
+      */
+      index = 100;
```
Comment (artem-blazhko) [inline src/UserSearchContainer.js:168]: Could you fix ESLint errors here?

**Review by artem-blazhko — COMMENTED** [review]
Could you cover new functionality by tests?

**Comment by manvendra-s-rathore** [comment]
> Could you cover new functionality by tests?

We have a tech debt to cover this later on. [UIPFU-78](https://folio-org.atlassian.net/browse/UIPFU-78)

**Review by artem-blazhko — APPROVED** [review]

**Review by BogdanDenis — APPROVED** [review]

**Pull request merged**